.github/workflows/ci.yml

```diff
@@ -51,24 +47,69 @@ jobs:
         - name: Build
           run: make all
 
+  lint:
+    if: (github.event.action != 'closed' || github.event.pull_request.merged == true)
+    strategy:
+      matrix:
+        os: [ ubuntu-20.04 ] # list of os: https://github.com/actions/virtual-environments
+    runs-on: ${{ matrix.os }}
+    steps:
+        - uses: actions/checkout@v3
+        - run: |
+            git submodule update --init --recursive --force
+            git fetch --no-tags --prune --depth=1 origin +refs/heads/master:refs/remotes/origin/master
+
+        - uses: actions/setup-go@v3
+          with:
+            go-version: 1.20.x
+
+        - name: Install dependencies on Linux
+          if: runner.os == 'Linux'
+          run: sudo apt update && sudo apt install build-essential
+
+        - name: Golang-ci install
+          if: runner.os == 'Linux'
+          run: make lintci-deps
+
         - name: Lint
           if: runner.os == 'Linux'
           run: make lint
 
+  unit-tests:
+    if: (github.event.action != 'closed' || github.event.pull_request.merged == true)
+    strategy:
+      matrix:
+        os: [ ubuntu-20.04 ] # list of os: https://github.com/actions/virtual-environments
+    runs-on: ${{ matrix.os }}
+    steps:
+        - uses: actions/checkout@v3
+        - run: |
+            git submodule update --init --recursive --force
+            git fetch --no-tags --prune --depth=1 origin +refs/heads/master:refs/remotes/origin/master
+
+        - uses: actions/setup-go@v3
+          with:
+            go-version: 1.20.x
+
+        - name: Install dependencies on Linux
+          if: runner.os == 'Linux'
+          run: sudo apt update && sudo apt install build-essential
+
+        - uses: actions/cache@v3
```

Comment:
I am pretty sure we need this in cache but it would be great if someone can confirm once?